spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -97,6 +97,14 @@ default boolean useOriginalInput() {
         return false;
     }
 
+    /**
+     * Whether the tool needs the history of the conversation.
+     * @return if true, agent runner should pass history information as input parameter to this tool
+     */
+    default boolean needHistory() {
```

Comment:
Adding new method to Tool interface is a concern to tool developers. They need to go through all methods to check if they need to implement the method or not. With more and more methods added to Tool interface, will take a lot of time to learn what's each method for, it also brings confusion. Let's discuss with @jngz-es to check if there are any other options ?

Reply:
@qianheng-aws Can you check if you can get chat history in tool by using `parameters.get("chat_history")`

Reply:
In CreatAlertTool, parameters don't have entry "chat_history" unless we pass the history to it. That's why we need this method to control whether passing "chat_history"

Reply:
Is this a necessary to add a new method to determine if the chat_history should be passed or not? Can we always pass chat_history to all tools as parameter so tool can access this based on their needs.

Reply:
@zane-neo  Incline with that if it's acceptable. What's your opinion? @ylwu-amzn